### PR TITLE
fix(elements): handle falsy initial value

### LINKED ISSUE #30834: Can't initialize Angular Element with a "falsy" element property value
# 🐞 bug report

### Affected Package
The issue is caused by package @angular/elements

### Is this a regression?
No.

### Description
When using an Angular Element as a custom element, a component `@Input` cannot be initialized to `false` (or other falsy value) via setting the element's property value. 

## 🔬 Minimal Reproduction
Stackblitz that shows failure to initialize both "false" (for boolean) and 0 (for number): https://stackblitz.com/edit/angular-elements-test-input-bindings

I debugged this and this `if` condition appears to be causing it ([component-factory-strategy.ts#L168](https://github.com/angular/angular/blob/master/packages/elements/src/component-factory-strategy.ts#L168)):
```
if (initialValue) {
    this.setInputValue(propName, initialValue);
} else {
    // Keep track of inputs that were not initialized in case we need to know this for
    // calling ngOnChanges with SimpleChanges
    this.uninitializedInputs.add(propName);
}
```

This is treating any "falsy" value as uninitialized, which is incorrect.

## 🔥 Exception or Error
There is no error, the boolean `false` input value is not set to the component input.

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 7.3.9
Node: 10.14.2
OS: darwin x64
Angular: 7.2.15
... animations, common, compiler, compiler-cli, core, elements
... forms, http, language-service, platform-browser
... platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.13.9
@angular-devkit/build-angular     0.13.9
@angular-devkit/build-optimizer   0.13.9
@angular-devkit/build-webpack     0.13.9
@angular-devkit/core              7.3.9
@angular-devkit/schematics        7.3.9
@angular/cdk                      7.2.2
@angular/cli                      7.3.9
@ngtools/webpack                  7.3.9
@schematics/angular               7.3.9
@schematics/update                0.13.9
rxjs                              6.4.0
typescript                        3.2.4
webpack                           4.29.0
</code></pre>

### PR DESCRIPTION
**Pull request by merobal**
Fixes angular/angular#30834

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: angular/angular#30834

## Does this PR introduce a breaking change?

- [ ] Yes
- [ ] No

### CODE REVIEW

**Comment by merobal** [comment]
I signed it!

File: packages/elements/src/component-factory-strategy.ts [hunk 1]
```diff
@@ -165,7 +165,7 @@ export class ComponentNgElementStrategy implements NgElementStrategy {
   protected initializeInputs(): void {
     this.componentFactory.inputs.forEach(({propName}) => {
       const initialValue = this.initialInputValues.get(propName);
-      if (initialValue) {
+      if (typeof initialValue !== 'undefined') {
```
Comment (gkalpak) [inline packages/elements/src/component-factory-strategy.ts]: What if the property is explicitly set to `undefined`? How about checking with `this.initialInputValue.has(propName)` instead?

Comment (merobal) [reply packages/elements/src/component-factory-strategy.ts]: Right, good idea. To achieve this, I had to put caching (https://github.com/angular/angular/pull/31604/commits/f4ccdf9a415ff62063c8bbe6cf736d5075190720#diff-c0d4cfddef0e50ed399624e91049d934R128) above checking the value via `strictEquals` to make sure we store it even if it's `undefined`. Please review and confirm if it's okay.

**Review by gkalpak — APPROVED** [review]
Thx, @merobal! Can you, please squash the two commits together?

**Comment by merobal** [comment]
@gkalpak done

**Comment by mhevery** [comment]
http://test/OCL:258798705:BASE:258798752:1563471301440:d3e1bcee

**Pull request closed without merging**